src/transforms/workflow.ts

```diff
@@ -33,42 +37,91 @@ function mapStatus(status: string, conclusion?: string): string {
-				id: pr.head.sha,
-				repositoryUri: pr.head.repo.url,
-			},
-			ref: {
-				name: pr.head.ref,
-				uri: `${pr.head.repo.url}/tree/${pr.head.ref}`,
-			},
-		}
-	));
+function mapPullRequests(
+	pull_requests: GitHubPullRequest[]
+): JiraPullRequest[] {
+	return pull_requests.map((pr) => ({
+		commit: {
+			id: pr.head.sha,
+			repositoryUri: pr.head.repo.url,
+		},
+		ref: {
+			name: pr.head.ref,
+			uri: `${pr.head.repo.url}/tree/${pr.head.ref}`,
+		},
+	}));
 }
 
-export default (context: Context): JiraBuildData | undefined => {
-	const { workflow_run, workflow } = context.payload;
-	const issueKeys = issueKeyParser().parse(`${workflow_run.head_branch}\n${workflow_run.head_commit.message}`);
+export default async (
+	githubClient: GitHubAPI,
+	payload: WorkflowPayload,
+	jiraHost: string,
+	logger?: LoggerWithTarget
+): Promise<JiraBuildData | undefined> => {
+	const { workflow_run, workflow } = payload;
+
+	const {
+		conclusion,
+		head_branch,
+		head_commit,
+		html_url,
+		name,
+		pull_requests,
+		repository,
+		run_number,
+		status,
+		updated_at,
+	} = workflow_run;
+
+	let issueKeys;
+
+	const supportBranchAndMergeWorkflowForBuildsFlagIsOn = await booleanFlag(BooleanFlags.SUPPORT_BRANCH_AND_MERGE_WORKFLOWS_FOR_BUILDS, false, jiraHost);
+	const workflowHasPullRequest = pull_requests.length > 0;
+
+	if (supportBranchAndMergeWorkflowForBuildsFlagIsOn && workflowHasPullRequest) {
+		const { owner, name: repoName } = repository;
+		const { base, head } = pull_requests[0];
+
+		const compareCommitsPayload = {
+			owner: owner.login,
+			repo: repoName,
+			base: base.ref,
+			head: head.ref
+		}
+
+		const allCommitMessages = await compareCommitsBetweenBaseAndHeadBranches(
+			compareCommitsPayload,
+			githubClient,
+			logger
+		);
+
+		issueKeys = issueKeyParser().parse(`${head_branch}\n${head_commit.message}\n${allCommitMessages}`) || [];
+	} else {
+		issueKeys =
+		issueKeyParser().parse(`${head_branch}\n${head_commit.message}`) || [];
+	}
 
 	if (!issueKeys) {
 		return undefined;
 	}
 
 	return {
 		product: "GitHub Actions",
-		builds: [{
-			schemaVersion: "1.0",
-			pipelineId: workflow.id,
-			buildNumber: workflow_run.run_number,
-			updateSequenceNumber: Date.now(),
-			displayName: workflow_run.name,
-			url: workflow_run.html_url,
-			state: mapStatus(workflow_run.status, workflow_run.conclusion),
-			lastUpdated: workflow_run.updated_at,
-			issueKeys,
-			references: workflow_run.pull_requests?.length ? mapPullRequests(workflow_run.pull_requests).slice(0, 5) : undefined // Optional information that links PRs. Min items: 1, Max items: 5
-		}],
+		builds: [
+			{
+				schemaVersion: "1.0",
+				pipelineId: workflow.id,
+				buildNumber: run_number,
+				updateSequenceNumber: Date.now(),
+				displayName: name,
+				url: html_url,
+				state: mapStatus(status, conclusion),
+				lastUpdated: updated_at,
+				issueKeys,
+				references: workflowHasPullRequest
+					? mapPullRequests(pull_requests).slice(0, 5)
```

Comment:
What's 5? Would it make sense to make it into a constant with a descriptive name?

Reply:
To be honest, I have no idea. I think this was added when we introduced builds and deployments in the app. I just changed `workflow_run.pull_requests` to pull_requests`

Reply:
Just realised there is already a comment there that says `// Optional information that links PRs. Min items: 1, Max items: 5`

Reply:
I'd suggest to make it into a constant e.g. `MAX_PULLREQUEST_REFERENCES` and then get rid of it